### PR TITLE
Fix inability to update grpc dependency to 1.55 or above by integrati…

### PR DESCRIPTION
**Pull request by DGuhr**
…ng parts of grpcutil

## Describe your changes

- Superseeds #133 by adding the part of the util we actually use in the infrastructure layer.
- Might be a temporary solution as long as https://github.com/authzed/grpcutil/issues/48 is not fixed. I'll keep an eye on that.

## Ticket reference (if applicable)
Fixes #

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [x] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [x] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: infrastructure/grpcutil/dial_options.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/certifi/gocertifi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
```
Comment (merlante) [inline infrastructure/grpcutil/dial_options.go:12]: I know we had these anyway, indirectly, but are we adding any new direct dependencies?

File: infrastructure/grpcutil/dial_options.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	return c, nil
+}
+
+// WithInsecureBearerToken returns a grpc.DialOption that adds a standard HTTP
+// Bearer token to all requests sent from an insecure client.
+//
+// Must be used in conjunction with `insecure.NewCredentials()`.
+func WithInsecureBearerToken(token string) grpc.DialOption {
+	return grpc.WithPerRPCCredentials(insecureMetadataCreds{"authorization": "Bearer " + token})
+}
```
Comment (merlante) [inline infrastructure/grpcutil/dial_options.go:86]: We are just replacing a library with the portion of the code that we need copied directly into our project, so LGTM.

**Review by merlante — APPROVED** [review]
I think it's good to go (although I am a little curious about the placement in infrastructure rather than api)!

File: infrastructure/grpcutil/dial_options.go (hunk 1 above)
Comment (DGuhr) [reply infrastructure/grpcutil/dial_options.go:12]: Feel free to resolve the conversation if that's no problem for you. :)

**Comment by DGuhr** [comment]
@merlante I added it to infra simply because the number of files where it's used is more (2) in infra than in api (1). Not at all hard on that :)

**Comment by DGuhr** [comment]
FYI: This change also removes the testify packages from the binary, as we are not using that part of grpcutil anymore.

 
Screenshot 2023-05-23 at 10 13 07

**Pull request merged**